pkg/agent/agent.go

```diff
@@ -396,7 +396,7 @@ func (i *Initializer) setupGatewayInterface() error {
 	gwIP := &net.IPNet{IP: ip.NextIP(subnetID), Mask: localSubnet.Mask}
 	gwAddr := &netlink.Addr{IPNet: gwIP, Label: ""}
 	gwMAC := link.Attrs().HardwareAddr
-	i.nodeConfig.GatewayConfig = &types.GatewayConfig{Link: i.hostGateway, IP: gwIP.IP, MAC: gwMAC}
+	i.nodeConfig.GatewayConfig = &types.GatewayConfig{Link: link, Name: i.hostGateway, IP: gwIP.IP, MAC: gwMAC}
```

Comment:
I find all the actual call of `types.GatewayConfig.Link` is to get the Link Index in node_route_controller, shall we only cache the index and name in the struct?

Reply:
makes sense, changed.